src/store/root-state.epics.ts

```diff
@@ -22,7 +24,11 @@ const rootStateResetEpic = (action$: Observable<Action>, state$: Observable<Root
     switchMap(keychainOptionsArray =>
       from(keychainOptionsArray).pipe(switchMap(options => Keychain.resetGenericPassword(options)))
     ),
-    mapTo(rootStateResetAction.success())
+    switchMap(() => from(BeaconHandler.getPermissions())),
+    switchMap(permissions =>
+      from(Promise.all(permissions.map(permission => BeaconHandler.removePermission(permission.accountIdentifier))))
+    ),
+    switchMap(() => [rootStateResetAction.success(), resetAllPermissionsAction()])
```

Comment:
you don't need `resetAllPermissionsAction` at all. `rootStateResetAction.success` will reset everything to the initial state (take look at `rootStateReducer`)